internal/namespaces/registry/v1/helpers_docker.go

```diff
@@ -0,0 +1,116 @@
+package registry
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+const (
+	dockerConfigDir      = ".docker"
+	dockerConfigFilename = "config.json"
+	dockerCredHelpersKey = "credHelpers"
+)
+
+func writeHelperScript(scriptPath string, scriptContent string) error {
+	scriptDirArg := path.Dir(scriptPath)
+	if _, err := os.Stat(scriptDirArg); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		if err := os.MkdirAll(scriptDirArg, 0755); err != nil {
+			return err
+		}
+	}
+
+	f, err := os.OpenFile(scriptPath, os.O_RDWR|os.O_CREATE, 0755)
```

Comment:
does O_CREATE implies O_TRUNC ?

Reply:
Nop but we don't want `O_TRUNC` here :thinking: